include/boost/geometry/algorithms/detail/sections/sectionalize.hpp

```diff
@@ -628,23 +624,21 @@ struct sectionalize_polygon
                 ring_identifier ring_id,
                 std::size_t max_count)
     {
-        typedef typename point_type<Polygon>::type point_type;
-        typedef sectionalize_range
+        using sectionalizer = sectionalize_range
             <
-                closure<Polygon>::value, Reverse, point_type, DimensionVector
-            > per_range;
+                closure<Polygon>::value, Reverse, DimensionVector
+            >;
 
         ring_id.ring_index = -1;
-        per_range::apply(exterior_ring(poly), robust_policy, sections,
+        sectionalizer::apply(exterior_ring(poly), robust_policy, sections,
                          strategy, ring_id, max_count);
 
         ring_id.ring_index++;
-        typename interior_return_type<Polygon const>::type
-            rings = interior_rings(poly);
-        for (typename detail::interior_iterator<Polygon const>::type
-                it = boost::begin(rings); it != boost::end(rings); ++it, ++ring_id.ring_index)
+        auto const& rings = interior_rings(poly);
```

Comment:
I think this is right, but I'm not 100% sure. 
Should it stay `interior_return_type` ?

Reply:
`const&` is fine because even if `interior_rings()` returned proxy reference like `reference_wrapper` this object would live as long as the const reference `rings`.

You could also use `auto&&` which would be able to represent both true references and proxy references exactly as they were returned from `interior_rings()`. But since we don't need perfect forwarding here enforcing constness is probably better.